main_test.go

```diff
@@ -17,15 +20,44 @@ func TestMain(m *testing.M) {
 	}))
 }
 
+func commonEnv(t *testing.T) func(env *testscript.Env) error {
+	return func(env *testscript.Env) error {
+		env.Setenv(string(v2.EnvDev), "true")
+		env.Setenv("HOME", "$WORK/home")
+		env.Setenv("VERSION", static.Version)
+		env.Setenv("COMMIT_SHA", static.Sha)
+		return nil
+	}
+}
+
 func TestScripts(t *testing.T) {
-	testscript.Run(t, testscript.Params{
-		Dir: "testdata/scripts",
-		Setup: func(env *testscript.Env) error {
-			env.Setenv(string(v2.EnvDev), "true")
-			env.Setenv("HOME", "$WORK/home")
-			env.Setenv("VERSION", static.Version)
-			env.Setenv("COMMIT_SHA", static.Sha)
-			return nil
-		},
-	})
+	t.Parallel()
+
+	testDataRootDir := "testdata"
+	testFs := os.DirFS(".")
+
+	visitFn := func(path string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			matches, err := fs.Glob(os.DirFS(path), "*txtar")
+			if err != nil {
+				return err
+			}
```

Comment:
Can this be simplified away by using an embed filter?
```go
//go:embed testdata/**/*.txtar
var f embed.FS

Reply:
doesn't seem like it.

testscript seems to fail if called on a dir with no txtar, so we need this matching.

and the glob abstraction takes a FS and a pattern if we only have the embedded fs, we would need to change the glob on each iteration.

Reply:
Why would the embed filter let through dirs w/o txtar files? Do we expect to have sub-dirs in the middle that only contain other dirs? I would think we have at least the basic `help.txtar` for each subcommand.